File: DSA Unit 05/Lec 31/Q1.cpp
Single File Programming Question
Problem Statement



﻿Arjun is working on a problem where he needs to find the length of the longest consecutive sequence of integers in an unsorted array. The sequence must consist of consecutive integers, and the order doesn't matter. Help Arjun implement this task efficiently.

Input format :
The first line of input consists of an integer n, representing the number of elements in the array.

The second line of input consists of n integers, representing the elements of the array.

Output format :
The first line of output prints: "Longest consecutive sequence length: " followed by the length of the longest consecutive subsequence in the array.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ n ≤ 1000
The array elements are integers that can range from -1000 to 1000.
MAX_SIZE = 1000
Sample test cases :
Input 1 :
6
100 4 200 1 3 2
Output 1 :
Longest consecutive sequence length: 4
Input 2 :
7
1 9 3 10 4 20 2
Output 2 :
Longest consecutive sequence length: 4


#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    unordered_set<int> s(arr.begin(), arr.end());
    int longest = 0;

    for (int x : arr) {
        if (s.find(x - 1) == s.end()) {
            int current = x;
            int length = 1;

            while (s.find(current + 1) != s.end()) {
                current++;
                length++;
            }

            longest = max(longest, length);
        }
    }

    cout << "Longest consecutive sequence length: " << longest;
    return 0;
}

File: Java/Unit 01/Static/Q1.java
Single File Programming Question
Problem Statement



Carlos is managing TechNova Electronics' inventory system. She is given a task to create a program that reads the device name, model number, and status and displays the formatted device information.



Since she is new to programming, can you assist Carlos in this task?

Input format :
The first line of input consists of a string, representing the device name.

The second line consists of a string, representing the status.

The third line consists of an integer, representing the model number.

Output format :
The first line of output prints "Device Name: " followed by a string representing the device name.

The second line prints "Status: " followed by a string representing the status.

The third line prints "Model Number: " followed by an integer representing the model number.



Refer to the sample output for formatting specifications.

Code constraints :
The given testcases fall under the following constraints:

The device name is a non-empty string containing only letters, digits, and spaces.

The status is a non-empty string and can include letters and spaces (e.g., "Available", "Out of Stock", "In Repair").

1 ≤ model number ≤ 1,000,000

Sample test cases :
Input 1 :
iPhone
Available
12
Output 1 :
Device Name: iPhone
Status: Available
Model Number: 12
Input 2 :
Galaxy
Out of Stock
10
Output 2 :
Device Name: Galaxy
Status: Out of Stock
Model Number: 10




import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String deviceName = sc.nextLine();
        String status = sc.nextLine();
        int modelNumber = sc.nextInt();

        System.out.println("Device Name: " + deviceName);
        System.out.println("Status: " + status);
        System.out.println("Model Number: " + modelNumber);

        sc.close();
    }
}


